src/app/app.cpp

```diff
@@ -163,9 +164,16 @@ int App::run(int argc, char** argv)
 
 #ifdef MUE_BUILD_APPSHELL_MODULE
     SplashScreen* splashScreen = nullptr;
+    NewInstanceLoadingProgressDialog* newInstanceLoadingDialog = nullptr;
     if (runMode == framework::IApplication::RunMode::GuiApp) {
-        splashScreen = new SplashScreen();
-        splashScreen->show();
+        if (multiInstancesProvider()->isMainInstance()) {
+            splashScreen = new SplashScreen();
+            splashScreen->show();
+        } else {
+            QString fileName = io::basename(startupScenario()->startupScorePath()).toQString();
+            newInstanceLoadingDialog = new NewInstanceLoadingProgressDialog(fileName);
+            newInstanceLoadingDialog->show();
```

Comment:
I do not claim, but it seems to me that it would be better to do only SplashScreen and it has different modes, i.e. here we create SplashScreen, set the mode and show it. And inside, depending on the mode, there are different implementations (we can do it within one widget or make SplashScreen as a facade that creates different widgets)